### DISCUSSION TITLE
Manual Layout Does Not Honor the Height Ratio for Rows

### DISCUSSION THREAD

**Discussion by RoyiAvital** [quarto-dev/quarto-cli#7082] [Q&A]
### Bug description

I use the layouts as documented in [Figures](https://quarto.org/docs/authoring/figures.html).

When I use something like:

```qmd
::: {layout="[[1], [1], [1]]"}

![](https://i.imgur.com/G2Y2TRd.png){fig-align="center"}

![](https://i.imgur.com/G2Y2TRd.png){fig-align="center"}

![](https://i.imgur.com/G2Y2TRd.png){fig-align="center"}

:::
```

I expect each row to have the same height and in total to occupy the height available.  
I practice it seems the height of the rows allocated is not independent of the content.

### Steps to reproduce

```qmd
---
format: revealjs
---

## Slide 1

::: {layout="[[-1], [1], [-1]]"}

![](https://i.imgur.com/G2Y2TRd.png){fig-align="center"}

:::

## Slide 2

::: {layout="[[2], [2], [1]]"}

![](https://i.imgur.com/G2Y2TRd.png){fig-align="center" width=50}

![](https://i.imgur.com/G2Y2TRd.png){fig-align="center" width=50}

![](https://i.imgur.com/G2Y2TRd.png){fig-align="center" width=50}

:::



```

### Expected behavior

 - Slide 1: The image should be centered in the slide. It should be limited to 33% of the height below the title.
 - Slide 2: The height of the area allocated to the first and second images to be twice the size of the last one.

### Actual behavior

 - Slide 1: The image stretched all over the slide:

![image](https://github.com/quarto-dev/quarto-cli/assets/7347975/91d14c8a-6b0c-4e60-9e1d-7fd6ee2e9b0b)

- Slide 2: The height allocated to each image is the same:

![image](https://github.com/quarto-dev/quarto-cli/assets/7347975/f50126e4-3592-43b1-b21c-f4f214c5fa30)

![image](https://github.com/quarto-dev/quarto-cli/assets/7347975/165e22c8-2418-4548-968b-63adcc92133d)

### Your environment

* IDE: Visual Studio 1.82.2.
* Quarto CLI 1.3.450.
* OS: Windows.

### Quarto check output

```cmd
[>] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.1: OK
      Dart Sass version 1.55.0: OK
[>] Checking versions of quarto dependencies......OK
[>] Checking Quarto installation......OK
      Version: 1.3.450
      Path: D:\Applications\Programming\Quarto\bin
      CodePage: 1252

[>] Checking basic markdown render....OK

[>] Checking Python 3 installation....OK

[>] Checking R installation...........(None)

      Unable to locate an installed version of R.
      Install R from https://cloud.r-project.org/
```

**Comment by cscheid** [comment]
Thanks for the report.

> I expect each row to have the same height and in total to occupy the height available.

Why do you expect that? Here's our documentation: https://quarto.org/docs/interactive/layout.html#panel-layout I don't think we promise constraints on height across rows, because enforcing each row to have the same height would make awkward layouts.

The numbers specify the relative width within any one row. Across rows, those numbers are not related to one another.

**Comment by RoyiAvital** [comment]
I see, so the only ratio given is about the width?
I was reading: https://stackoverflow.com/questions/72941210.
It seems other expected that as well.

Could we somehow also force the height to have a certain ratio?

**Comment by cscheid** [comment]
> Could we somehow also force the height to have a certain ratio?

I don't think there's a way to do it in the YAML cell metadata. It would be very hard to be able to control both width and height ratios, and then have to decide how to solve an overconstrained problem.

In these situations, your best bet is unfortunately to inspect the HTML and try to write CSS that solves the problem for you.
